alt-tab-macos/logic/PreferencesPanelNSTextFieldEditable.swift

```diff
@@ -0,0 +1,34 @@
+import Cocoa
+
+class PreferencesPanelNSTextFieldEditable: NSTextField, NSTextFieldDelegate {
+
+    var validationHandler: ((String)->Bool)?
+
+    // protocol method
+    func controlTextDidChange(_ obj: Notification) {
+        visualizeValidationState(isValid())
+        let textField = obj.object as! PreferencesPanelNSTextFieldEditable
+        sendAction(textField.action, to: textField.target)
+    }
+
+    // custom method
+    func visualizeValidationState(_ isValid: Bool) -> Void {
+        if !isValid {
+            wantsLayer = true
+            layer?.borderColor = NSColor.systemRed.cgColor
+            layer?.borderWidth = 1
+        } else {
+            wantsLayer = false
```

Comment:
I think this logic here is not correct. `wantsLayer` live on some lifecycle that I don't think come into play instantly here. That means if it becomes red, then goes into the `else`, it will not clear the red. I think do the same as I did for the thumbnails border: set `wantsLayer = true` on init always, then set either `.systemRed` or `.clear` in the `if/else`

Reply:
Thank you, corrected.